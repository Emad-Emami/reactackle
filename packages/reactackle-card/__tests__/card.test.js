'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import {
    Card,
    CardActions,
    CardAreaMain,
    CardAreaSide,
    CardContent,
    CardHeaderPrimary,
    CardHeaderSecondary,
    CardMedia,
    CardContentText,
    CardActionsMain,
    CardActionsSupplemental,
} from '../src';

jest.mock('react-dom');

describe('<Card/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Card>
        <div> test </div>
      </Card>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if contains CardAreaSide', () => {
    const tree = renderer.create(
      <Card>
        <CardAreaSide />
      </Card>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly if prop size changing', () => {
    const component = renderer.create(
      <Card size="large">
        <CardContent />
      </Card>,
    );

    expect(component.toJSON()).toMatchSnapshot();
    
    const instance = component.getInstance();
    instance.componentWillReceiveProps({ size: 'xlarge' });
    
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('<CardActions/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardActions>
        <div> test </div>
      </CardActions>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardAreaMain/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardAreaMain>
        <div> test </div>
      </CardAreaMain>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardAreaSide/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardAreaSide>
        <div> test </div>
      </CardAreaSide>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardContent/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardContent>
        <div> test </div>
      </CardContent>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop last set true', () => {
    const tree = renderer.create(
      <CardContent last />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop last set false', () => {
    const tree = renderer.create(
      <CardContent last={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('<CardHeaderPrimary/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardHeaderPrimary>
        <div> test </div>
      </CardHeaderPrimary>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop title set test', () => {
    const tree = renderer.create(
      <CardHeaderPrimary title="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop subtitle set test', () => {
    const tree = renderer.create(
      <CardHeaderPrimary subtitle="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('<CardHeaderSecondary/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardHeaderSecondary>
        <div> test </div>
      </CardHeaderSecondary>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop title set test', () => {
    const tree = renderer.create(
      <CardHeaderSecondary title="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop subtitle set test', () => {
    const tree = renderer.create(
      <CardHeaderSecondary subtitle="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop image set image.png', () => {
    const tree = renderer.create(
      <CardHeaderSecondary image="./image.png" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('<CardMedia/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardMedia>
        <div> test </div>
      </CardMedia>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardActionsSupplemental/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardActionsSupplemental>
        <div> test </div>
      </CardActionsSupplemental>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardActionsMain/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardActionsMain>
        <div> test </div>
      </CardActionsMain>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<CardContentText/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CardContentText>
        <div> test </div>
      </CardContentText>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
