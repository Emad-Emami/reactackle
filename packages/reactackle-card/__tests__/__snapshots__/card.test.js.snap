// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders correctly if contains CardAreaSide 1`] = `
<div
  className="CardStyled-egOvwJ eMIsCX"
  elevation={1}
>
  <div
    className="CardAreaSideStyled-idUIJJ iEcodt"
  />
</div>
`;

exports[`<Card/> renders correctly if prop size changing 1`] = `
<div
  className="CardStyled-egOvwJ eMIsCX"
  elevation={1}
>
  <div
    className="CardContentStyled-eXpgEI bCCZWM"
    size="large"
  />
</div>
`;

exports[`<Card/> renders correctly if prop size changing 2`] = `
<div
  className="CardStyled-egOvwJ eMIsCX"
  elevation={1}
>
  <div
    className="CardContentStyled-eXpgEI bCCZWM"
    size="xlarge"
  />
</div>
`;

exports[`<Card/> renders correctly with default props 1`] = `
<div
  className="CardStyled-egOvwJ eMIsCX"
  elevation={1}
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardActions/> renders correctly with default props 1`] = `
<div
  className="CardActionsStyled-iojRbO kinqsb"
  size="default"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardActionsMain/> renders correctly with default props 1`] = `
<div
  className="CardActionsMainStyled-eJCwgY XsMdB"
  size="default"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardActionsSupplemental/> renders correctly with default props 1`] = `
<div
  className="CardActionsSupplementalStyled-bZhMlP fBwYYd"
  size="default"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardAreaMain/> renders correctly with default props 1`] = `
<div
  className="CardAreaMainStyled-bhXci cgbqBs"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardAreaSide/> renders correctly with default props 1`] = `
<div
  className="CardAreaSideStyled-idUIJJ iEcodt"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardContent/> renders correctly if prop last set false 1`] = `
<div
  className="CardContentStyled-eXpgEI iegvJK"
  size="default"
/>
`;

exports[`<CardContent/> renders correctly if prop last set true 1`] = `
<div
  className="CardContentStyled-eXpgEI dlppKT"
  size="default"
/>
`;

exports[`<CardContent/> renders correctly with default props 1`] = `
<div
  className="CardContentStyled-eXpgEI iegvJK"
  size="default"
>
  <div>
     test 
  </div>
</div>
`;

exports[`<CardContentText/> renders correctly with default props 1`] = `
<p
  className="CardContentTextStyled-ihOUSW gaxXsB"
  size="default"
>
  <div>
     test 
  </div>
</p>
`;

exports[`<CardHeaderPrimary/> renders correctly if prop subtitle set test 1`] = `
<div
  className="CardHeaderPrimaryStyled-hSQWwz laFTcJ"
  size="default"
>
  <div
    className="CardHeaderBoxStyled-eqowYz hOKnNi"
  >
    <h2
      className="CardHeaderTextStyled-OYREa iixgon"
      size="default"
    >
      
    </h2>
    <div
      className="CardSubheaderStyled-hGLFKT eHmZDq"
      size="default"
    >
      test
    </div>
  </div>
</div>
`;

exports[`<CardHeaderPrimary/> renders correctly if prop title set test 1`] = `
<div
  className="CardHeaderPrimaryStyled-hSQWwz laFTcJ"
  size="default"
>
  <div
    className="CardHeaderBoxStyled-eqowYz hOKnNi"
  >
    <h2
      className="CardHeaderTextStyled-OYREa iixgon"
      size="default"
    >
      test
    </h2>
  </div>
</div>
`;

exports[`<CardHeaderPrimary/> renders correctly with default props 1`] = `
<div
  className="CardHeaderPrimaryStyled-hSQWwz laFTcJ"
  size="default"
>
  <div
    className="CardHeaderBoxStyled-eqowYz hOKnNi"
  >
    <h2
      className="CardHeaderTextStyled-OYREa iixgon"
      size="default"
    >
      
    </h2>
  </div>
  <div
    className="CardHeaderMediaStyled-jZvFHX pGezn"
    size="default"
  >
    <div>
       test 
    </div>
  </div>
</div>
`;

exports[`<CardHeaderSecondary/> renders correctly if prop image set image.png 1`] = `
<div
  className="CardHeaderSecondaryStyled-bqsWeH KrrET"
  size="default"
>
  <div
    className="CardHeaderSecondaryImageStyled-gJSkO dWyfnh"
    size="default"
    style={
      Object {
        "backgroundImage": "url('./image.png')",
      }
    }
  />
  <div
    className="CardHeaderSecondaryTextBoxStyled-ejWykG iklcYU"
  >
    <div
      className="CardHeaderSecondaryTextStyled-kDiNLK kLBBdu"
      size="default"
    >
      
    </div>
  </div>
</div>
`;

exports[`<CardHeaderSecondary/> renders correctly if prop subtitle set test 1`] = `
<div
  className="CardHeaderSecondaryStyled-bqsWeH KrrET"
  size="default"
>
  <div
    className="CardHeaderSecondaryTextBoxStyled-ejWykG iklcYU"
  >
    <div
      className="CardHeaderSecondaryTextStyled-kDiNLK kLBBdu"
      size="default"
    >
      
    </div>
    <div
      className="CardSubheaderSecondaryStyled-lfDxSo kKdifI"
      size="default"
    >
      test
    </div>
  </div>
</div>
`;

exports[`<CardHeaderSecondary/> renders correctly if prop title set test 1`] = `
<div
  className="CardHeaderSecondaryStyled-bqsWeH KrrET"
  size="default"
>
  <div
    className="CardHeaderSecondaryTextBoxStyled-ejWykG iklcYU"
  >
    <div
      className="CardHeaderSecondaryTextStyled-kDiNLK kLBBdu"
      size="default"
    >
      test
    </div>
  </div>
</div>
`;

exports[`<CardHeaderSecondary/> renders correctly with default props 1`] = `
<div
  className="CardHeaderSecondaryStyled-bqsWeH KrrET"
  size="default"
>
  <div
    className="CardHeaderSecondaryTextBoxStyled-ejWykG iklcYU"
  >
    <div
      className="CardHeaderSecondaryTextStyled-kDiNLK kLBBdu"
      size="default"
    >
      
    </div>
  </div>
</div>
`;

exports[`<CardMedia/> renders correctly with default props 1`] = `
<div
  className="CardMediaStyled-cUEpZl bSWiJZ"
>
  <div
    className="CardMediaContentStyled-cENUeV jLwMXl"
  >
    <div>
       test 
    </div>
  </div>
</div>
`;
